### PR TITLE
Convert picture display to a phase

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Refactors the phasers to be a bit more generic and robust.
I'm not sure if we're not eventually going to need a stack for the phasers.

### CODE REVIEW

File: src/game/game/game_cutscene.c [hunk 1]
```diff
@@ -56,10 +56,10 @@ bool Game_Cutscene_Control(int32_t nframes)
     return false;
 }
 
-int32_t Game_Cutscene_Start(int32_t level_num)
+GAMEFLOW_OPTION Game_Cutscene_Start(int32_t level_num)
 {
     if (!Level_Initialise(level_num)) {
-        return END_ACTION;
```
Comment (walkawayy) [inline src/game/game/game_cutscene.c:62]:
```
#define NO_ACTION 0
#define GAME_ACTION 1
#define END_ACTION 2
#define LOAD_ACTION 4
#define SAVE_ACTION 8
#define SOUND_ACTION 16
#define VIDEO_ACTION 32
```

Are all unused now and can be deleted from `const.h`.

File: src/game/game/game_demo.c [hunk 2]
```diff
@@ -40,6 +45,14 @@ static void Game_Demo_LoadLaraPos(void)
     FLOOR_INFO *floor =
         Room_GetFloor(item->pos.x, item->pos.y, item->pos.z, &room_num);
     item->floor = Room_GetHeight(floor, item->pos.x, item->pos.y, item->pos.z);
+
+    Random_SeedDraw(0xD371F947);
+    Random_SeedControl(0xD371F947);
+
+    // LaraGun() expects request_gun_type to be set only when it
+    // really is needed, not at all times.
+    // https://github.com/LostArtefacts/TR1X/issues/36
```
Comment (walkawayy) [inline src/game/game/game_demo.c:54]: I think this comment about a super old, closed issue can be deleted? I don't think it's still relevant.

**Review by walkawayy — APPROVED** [review]
No issues I noticed with pics on startup or credits, demo mode, normal game, cutscenes, or FMVs. LGTM.

File: src/game/game/game_demo.c (hunk 2 above)
Comment (rr-) [reply src/game/game/game_demo.c:54]: It still is relevant:
1. Run normal game
2. Equip weapons
3. Use the pause screen (or likely the inventory screen) to exit to the main menu
4. Wait for the demo to run
5. Lara equips her pistols at the start of the sequence

**Pull request merged**